slamd/design_assistant/static/design_knowledge.js

```diff
@@ -1,59 +1,46 @@
 import {scrollDown, updateProgress} from "./utils.js";
 
-export function assignEventsToDesignKnowledge() {
-    assignClickEventToGenerateDesignKnowledgeButton()
-    assignClickEventToContinueButton()
+import { assignClickEventToSubmitButton } from "./utils.js"
+import { assignEventsToFormulation } from "./formulation.js";
+
+export function assignEventsToDesignKnowledgeForm() {
+    assignClickEventToSubmitButton("generate_design_knowledge_button", handleGeneratingDesignKnowledge)
+    assignClickEventToSubmitButton("generate_formulation_button", handleGeneratingFormulation)
 }
 
-export function assignClickEventToGenerateDesignKnowledgeButton() {
-    const generate_design_knowledge_button = document.getElementById("generate_design_knowledge_button")
-    if (generate_design_knowledge_button) {
-        generate_design_knowledge_button.addEventListener('click', async function () {
-            insertSpinnerInPlaceholder(
-                "llm_response_container",
-                true,
-                CHATBOT_RESPONSE_SPINNER
-            );
-            setTimeout(async function handleSubmission() {
-                await postDataAndEmbedTemplateInPlaceholder(
-                    "/design_assistant/zero_shot/generate_design_knowledge",
-                    "llm_response",
-                    {"token": document.getElementById("token_form-token").value}
-                );
-                removeSpinnerInPlaceholder("llm_response_container", CHATBOT_RESPONSE_SPINNER)
-                document.getElementById("llm_response").classList.remove('d-none')
-            }, 1000);
-            document.getElementById("continue_design_knowledge_button").disabled = false
-            document.getElementById("llm_response").disabled = false
+async function handleGeneratingDesignKnowledge(){
+    insertSpinnerInPlaceholder("design_knowledge_inner_container",true,CHATBOT_RESPONSE_SPINNER);
+    setTimeout(async function handleSubmission() {
+        await postDataAndEmbedTemplateInPlaceholder(
+            "/design_assistant/zero_shot/generate_design_knowledge",
+            "design_knowledge",
+            {"token": document.getElementById("token_form-token").value}
+        );
+        removeSpinnerInPlaceholder("design_knowledge_inner_container", CHATBOT_RESPONSE_SPINNER)
```

Comment:
Muss hier noch das

    updateProgress()
    scrollDown()
    
   gemacht werden.
   
   (Bsp dazu findest du in den anderen js files)